aten/src/ATen/native/Activation.cpp

```diff
@@ -41,11 +41,11 @@ Tensor & celu_(Tensor & self, Scalar alpha) {
 }
 
 Tensor rrelu(const Tensor & self, Scalar lower, Scalar upper, bool training, Generator* generator) {
-  return at::rrelu_with_noise(self, at::empty({0}, self.options()), lower, upper, training, generator);
+  return at::rrelu_with_noise(self, at::empty_like(self), lower, upper, training, generator);
 }
 
 Tensor & rrelu_(Tensor & self, Scalar lower, Scalar upper, bool training, Generator* generator) {
-  return at::rrelu_with_noise_(self, at::empty({0}, self.options()), lower, upper, training, generator);
+  return at::rrelu_with_noise_(self, at::empty_like(self), lower, upper, training, generator);
```

Comment:
We need these two changes for `rrelu` to unbreak its backward path, because of the following reason:

The forward path of `rrelu` will resize the `noise` tensor (the tensor we are creating with `at::empty_like()` here), which will change its size/strides information. Before this patch, this behavior is fine because SavedVariable actually saves the underlying data tensor, and if the underlying data tensor's size/strides is changed, the SavedVariable is updated accordingly. However, after the patch, SavedVariable actually saves a shallow copy of the Variable and does not share the size/strides, and if we change the size/strides of the `noise` tensor after it's saved, the SavedVariable has no way to know about it, and the saved tensor inside will be wrong.

The way to fix this problem is to set the size of the `noise` tensor appropriately before passing it into the forward function, which is what we are doing here.

Reply:
Tracking issue is at https://github.com/pytorch/pytorch/issues/17506 (thanks Greg ;).